tests/php/Analyzer/TupleSymbol/FnSymbolTest.php

```diff
@@ -0,0 +1,178 @@
+
+    public function testSecondArgMustBeATuple(): void
+    {
+        $this->expectException(PhelCodeException::class);
+        $this->expectExceptionMessage("Second argument of 'fn must be a Tuple");
+
+        // This is the same as: (anything anything)
+        $tuple = Tuple::create(
+            Symbol::create('anything'),
+            Symbol::create('anything')
+        );
+
+        $this->analyze($tuple);
+    }
+
+    public function testIsVariadic(): void
+    {
+        // This is the same as: (anything (anything))
+        $tuple = Tuple::create(
+            Symbol::create('anything'),
+            Tuple::create(Symbol::create('anything'))
+        );
+
+        $fnNode = $this->analyze($tuple);
+
+        self::assertFalse($fnNode->isVariadic());
+    }
+
+    /**
+     * @dataProvider providerVarNamesMustStartWithLetterOrUnderscore
+     */
+    public function testVarNamesMustStartWithLetterOrUnderscore(string $paramName, bool $error): void
+    {
+        if ($error) {
+            $this->expectException(PhelCodeException::class);
+            $this->expectExceptionMessageMatches('/(Variable names must start with a letter or underscore)*/i');
+        } else {
+            self::assertTrue(true); // In order to have an assertion without an error
+        }
+
+        // This is the same as: (anything (fn [paramName]))
+        $tuple = Tuple::create(
+            Symbol::create('anything'),
+            Tuple::create(
+                Symbol::create(Symbol::NAME_FN),
+                Symbol::create($paramName)
+            ),
+        );
+
+        $this->analyze($tuple);
+    }
+
+    public function providerVarNamesMustStartWithLetterOrUnderscore(): Generator
+    {
+        yield 'Start with a letter' => [
+            'paramName' => 'param-1',
+            'error' => false,
+        ];
+
+        yield 'Start with an underscore' => [
+            'paramName' => '_param-2',
+            'error' => false,
+        ];
+
+        yield 'Start with a number' => [
+            'paramName' => '1-param-3',
+            'error' => true,
+        ];
+
+        yield 'Start with an ampersand' => [
+            'paramName' => '&-param-4',
+            'error' => true,
+        ];
+
+        yield 'Start with a space' => [
+            'paramName' => ' param-5',
+            'error' => true,
+        ];
+    }
+
+    public function testOnlyOneSymbolCanFollowTheAmpersandParameter(): void
+    {
+        $this->expectException(PhelCodeException::class);
+        $this->expectExceptionMessage('Unsupported parameter form, only one symbol can follow the & parameter');
+
+        // This is the same as: (anything (fn [& param-1 param-2]))
+        $tuple = Tuple::create(
+            Symbol::create('anything'),
+            Tuple::create(
+                Symbol::create(Symbol::NAME_FN),
+                Symbol::create('&'),
+                Symbol::create('param-1'),
+                Symbol::create('param-2'),
+            ),
+        );
```

Comment:
This is not `(anything (fn [& param-1 param-1))` but `(anything (fn & param-1 param-2))`. The test should be like this:
```
$tuple = Tuple::create(
  Symbol::create(Symbol::NAME_FN),
  Tuple::createBracket(
    Symbol::create('&'),
    Symbol::create('param-1'),
    Symbol::create('param-2'),
  )
)
```
Which is equal to `(fn [& param-1 param-2])`.